zkevm-circuits/src/test_util.rs

```diff
@@ -175,12 +188,19 @@ impl<const NACC: usize, const NTX: usize> CircuitTestBuilder<NACC, NTX> {
 
         let (active_gate_rows, active_lookup_rows) = EvmCircuit::<Fr>::get_active_rows(&block);
 
-        let circuit = EvmCircuitCached::get_test_circuit_from_block(block);
-        let prover = MockProver::<Fr>::run(k, &circuit, vec![]).map_err(|err| {
-            CircuitTestError::SynthesisFailure {
-                circuit: Circuit::EVM,
-                reason: err,
-            }
+        // Mainnet EVM circuit constraints can be cached for test performance.
+        // No cache for EVM circuit with customized features
+        let prover = if block.feature_config.is_mainnet() {
```

Comment:
Very nice! This resolves my question about using the cached version in non-mainnet features :)